client/modules/transactions.js

```diff
@@ -38,9 +37,10 @@ function getNextTransactionID(state) {
 
 // Add a new transaction.
 function addTransactionToState(state, action) {
-  const { description, value } = action.transaction;
+  const { categoryId, description, value } = action.transaction;
   const newState = [...state, {
     id: getNextTransactionID(state),
+    categoryId,
```

Comment:
or `...action.transaction` and then we can get rid of this line and the next two ?